### PR TITLE
Update gitignore

### PR DESCRIPTION
**Pull request by cdce8p**
## Steps

- [ ] Add yourself to CONTRIBUTORS if you are a new contributor.
- [ ] Add a ChangeLog entry describing what your PR does.
- [ ] If it's a new feature or an important bug fix, add a What's New entry in `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.

## Description
Update `gitingore` file with some common files and folders.
Those are just some I frequently have to exclude.

Although not strictly necessary, it could make sense to use some more extensive templates like:
https://github.com/github/gitignore
https://www.toptal.com/developers/gitignore?templates=python,windows,macos,vscode

## Type of Changes

|   | Type |
| ------------- | ------------- |
|   | :bug: Bug fix  |
|   | :sparkles: New feature |
|   | :hammer: Refactoring  |
|   | :scroll: Docs |

### CODE REVIEW

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
We started to add things not directly linked to pylint in here at some point so it makes sense to want to keep adding to it. For this reason this is becoming something we have to maintain and doing this one merge at a time is time consuming. I think we have two options at this point:
- Going all in with a generic `.gitignore` that should handle all major IDE/environment so it's "done"
- Or, going back to just handling pylint's stuff and putting documentation for the global .gitignore in.

 I think the second solution is better in the long term. pylint's gitignore would be smaller, each developer global gitignore too, and it will help them have smaller gitignore in their other projects. Also if another cool IDE appear we won't have to make this evolve.
 
 What do you think ?

File: .gitignore [hunk 1]
```diff
@@ -25,3 +25,7 @@ debian/pylint
 .eggs/
 .pytest_cache/
 .mypy_cache/
+venv*/
+.vscode/
```
Comment (PCManticore) [inline .gitignore:29]: These are not Pylint specific but they are specific to your dev environment or development preferences. This `.gitignore` currently contains only ignores specific to the development of `pylint`.

**Comment by cdce8p** [comment]
To be honest, I have used git for a few years now and never knew about the global gitignore. I only found it a few days ago shortly after opening this MR. Most Open Source projects I have contributed to in the past, just use one of the templates and that works really well. I understand your point that you don't want to have to maintain it, but a good template goes a long way without needing much change. Especially for beginners with little git experience, it makes things just easier. Although I suppose, the overwhelming majority of pylint contributors aren't beginners :)

**Comment by Pierre-Sassoulas** [comment]
I opened #4076 following this.

**Pull request closed without merging**